### DISCUSSION TITLE
PR Review Comments Unresolvable By Owner of Review

### DISCUSSION THREAD

**Discussion by nrmancuso** [community/community#8327] [Pull Requests]
From the github documentation at https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews:

`You can resolve a conversation in a pull request if you opened the pull request or if you have write access to the repository where the pull request was opened.`

As a collaborator with triage permissions and maintainer of an open source project, I should be able to resolve my own review items even though I do not have write access to the repository. I can delete my own review items, edit them, etc., so it follows that I should be able to resolve my own review items.

Example of needing member with read/ write access to resolve items: https://github.com/checkstyle/checkstyle/pull/10990#pullrequestreview-818596050

To me - having the ability to delete review items where there could be discussion from other contributors and organization members has the potential to be much more destructive than simply resolving my own review items.

**Comment by romani** [comment]
just a comment .... To make it even more weird:
Author of PR can resolve conversation created by reviewer with Triage permission in that repo, BUT reviewer can not.
But it is good practice to let author of review point to resolve it as confirmation that all is done and point can be collapsed.

**Comment by Vyom-Yadav** [comment]
Agreed such a feature would be really helpful. Unable to resolve review items on my own creates a lot of noise in the comments and it is difficult to navigate through the conversation. Many times there are multiple unresolved reviews that have been addressed and should be in the resolved state but require the intervention of someone with write access to the repo which makes reviewing tedious.

**Comment by ES-Alexander** [comment]
Per #187503, the interface provides a "Resolve" button in a couple of different places that raises a 401 error when clicked:

<img width="400" height="432" alt="Screenshot 2026-02-24 at 5 51 17 pm" src="https://github.com/user-attachments/assets/9bb2ed6b-8544-4cb9-aa2e-68f45c397ef9" />   
<img width="400" height="480" alt="Screenshot 2026-02-24 at 5 51 58 pm" src="https://github.com/user-attachments/assets/dfdcb139-14b0-402a-9227-6d3fb08052a8" />

so to me this feels like a (confusing and frustrating) bug rather than an intentional permissions limitation. The documentation is correct for the UX (of who ends up being allowed to resolve threads), but is inconsistent with the UI (and some contributors with good reason to believe they should be able to resolve their own threads).

**Comment by mikebeaton** [comment]
It does seem natural that a reviewer should be able to resolve their own comments - if anything more so than the creator of the PR.

As others mentioned (but over four years ago, so perhaps it's okay to repeat), it does create noise and prevents the otherwise good practice that the person who raises an issue marks it resolved when it is resolved (so that everyone else doesn't have to keep reading and re-reading it). And as @nrmancuso pointed out, a reviewer can often do seemingly more 'powerful'/potentially harmful things, even while not being able to mark as resolved their own comments.

The OP and comments have got quite a few upvotes - is there any chance of progress/change on this?
